Office 365/message-trace/ingest/parser.yml

```diff
@@ -0,0 +1,49 @@
+name: message-trace
+pipeline:
+  - name: json_event
+    external:
+      name: json.parse-json
+      properties:
+        input_field: "{{original.message}}"
+
+  # Example of date field # '\\/Date(1658579297628)\\/'
+  - name: parse_start_date
+    external:
+      name: date.parse
+      properties:
+        input_field: '{{json_event.message.StartDate | replace("\\/Date(","") | replace(")\\/","") | int / 1000}}'
+        output_field: datetime
+        format: 'timestamp'
+  - name: parse_end_date
+    external:
+      name: date.parse
+      properties:
+        input_field: '{{json_event.message.EndDate | replace("\\/Date(","") | replace(")\\/","") | int / 1000}}'
+        output_field: datetime
+        format: 'timestamp'
+  - name: parse_received_date
+    external:
+      name: date.parse
+      properties:
+        input_field: '{{json_event.message.EndDate | replace("\\/Date(","") | replace(")\\/","") | int / 1000}}'
+        output_field: datetime
+        format: 'timestamp'
+
+  - name: extract_fields
+
+stages:
+  extract_fields:
+    actions:
+      - set:
+          office365.message_trace.MessageTraceId: "{{json_event.message.MessageTraceId}}"
+          office365.message_trace.Size: "{{json_event.message.Size}}"
+          office365.message_trace.Status: "{{json_event.message.Status}}"
+          event.start: "{{parse_start_date.datetime}}"
+          event.end: "{{parse_end_date.datetime}}"
+          email.delivery_timestamp: "{{parse_received_date.datetime}}"
+          source.ip: "{{json_event.message.FromIP}}"
+          email.message_id: "{{json_event.message.MessageId}}"
+          email.sender.address: "{{json_event.message.SenderAddress}}"
```

Comment:
According to the meaning of `SenderAddress`, maybe set this value to `email.from.address`?